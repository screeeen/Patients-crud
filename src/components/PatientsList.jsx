import React from 'react'
import { Link } from "react-router-dom";

const PatientsList = ({ patientList, editPatient, deletePatient, addNewPatient }) => {
  return (
    <>
      <table>
        <thead>
          <tr>
            <th>Id</th>
            <th>Name</th>
            <th>Surname</th>
            <th>Age</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {
            patientList.length > 0 ? (
              patientList.map(patient => (
                <tr key={patient.id}>
                  <td>{patient.id}</td>
                  <td>{patient.name}</td>
                  <td>{patient.surname}</td>
                  <td>{patient.age}</td>
                  <td><button onClick={() => { editPatient(patient) }}>edit</button></td>
                  <td><button onClick={() => { deletePatient(patient.id) }}>delete</button></td>
                </tr>
              )
              )
            ) : (
                <tr>
                  <td>No patients yet.</td>
                </tr>
              )
          }
        </tbody>
      </table >
      <Link to="/add">Register new patient</Link>
    </>
  )
}

export default PatientsList


