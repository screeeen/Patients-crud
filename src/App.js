import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from "react-router-dom";
import AddPatient from './components/AddPatient'
import EditPatient from './components/EditPatient'
import PatientsList from './components/PatientsList'
import './index.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome } from '@fortawesome/free-solid-svg-icons'


class App extends Component {

  constructor() {
    super();
    this.state = {
      id: undefined,
      patient: {},
      patientList: [],
    }

    this.addNewPatient = this.addNewPatient.bind(this);
    this.deletePatient = this.deletePatient.bind(this);
    this.editPatient = this.editPatient.bind(this);
  }


  mockData = [
    { id: 0, name: 'Manuel', surname: 'Pereira', age: 30, birth: new Date(1989, 1) },
    { id: 1, name: 'Eddo', surname: 'Park', age: 51, birth: new Date(1968, 1) },
    { id: 2, name: 'Clark', surname: 'Syntherion', age: 45, birth: new Date(1974, 1) },
    { id: 3, name: 'Aphex', surname: 'Singlar', age: 90, birth: new Date(1930, 1) },
  ]

  componentDidMount() {
    this.setState({ patientList: this.mockData });
  }

  getLastId() {
    if (this.state.patientList.length - 1 <= 0) return 0;
    const lastIndex = this.state.patientList.length - 1;
    return this.state.patientList[lastIndex].id + 1;
  }

  addNewPatient(patient) {
    patient.id = this.getLastId();
    this.setState({ patientList: [...this.state.patientList, patient] });
  }

  editPatient(updatedPatient) {
    const patientList = this.state.patientList.map((patient) => (patient.id === updatedPatient.id ? updatedPatient : patient));
    this.setState({ patientList: patientList });
  }

  deletePatient(id) {
    const patientList = this.state.patientList.filter(patient => patient.id !== id);
    this.setState({ patientList: patientList });
    if (this.state.editing === true) {
      window.location.reload();
    }
  }

  render() {
    const { patientList } = this.state;
    return (
      <div className='container'>
        <Router>
          <nav >
            <ul>
              <li>
                <Link to="/"><p><FontAwesomeIcon icon={faHome} size='lg'/></p></Link>
              </li>
              {/* <li>
                <Link to="/add">Add</Link>
              </li> */}
              {/* <li>
                <Link to="/edit">Edit</Link>
              </li> */}
            </ul>
          </nav>
          <Switch>
            <Route exact path="/" component={() => <PatientsList patientList={patientList} editPatient={this.editPatient} deletePatient={this.deletePatient} addNewPatient={this.addNewPatient} />} />
            <Route exact path="/add" component={() => <AddPatient addNewPatient={this.addNewPatient} />} />
            <Route exact path="/edit" component={() => <EditPatient patient={this.state.patient} editPatient={this.editPatient} />} />
          </Switch>
        </Router>
      </div>
    );
  }
}

export default App;
