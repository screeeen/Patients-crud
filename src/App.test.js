import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import PatientsList from './components/PatientsList'

it('list is displayed', () => {
const list = (<PatientsList />)

  expect(list).toMatchSnapshot();
});

it('list displays message if is empty', () => {
  // const list = (<PatientsList />)
  // expect(user.age).
});


it('user cant be less than 0 years old', () => {
  const user = {
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
    surname :'paqueton',
    age:2,
    birth:new Date()
  };
  expect(user.age).toBeGreaterThan(0);
});


