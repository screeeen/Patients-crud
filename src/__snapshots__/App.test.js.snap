// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list is displayed 1`] = `<withRouter(PatientsList) />`;

exports[`user can be created 1`] = `
Object {
  "age": 45,
  "birth": 2020-04-17T01:55:46.458Z,
  "id": 16,
  "name": "LeBron James",
  "surname": "paqueton",
}
`;

exports[`user can be decribed 1`] = `
Object {
  "age": 45,
  "birth": 2020-04-17T01:57:00.926Z,
  "id": 8,
  "name": "LeBron James",
  "surname": "paqueton",
}
`;

exports[`user cant be less than 0 years old 1`] = `
Object {
  "age": -2,
  "birth": 2020-04-17T01:57:00.930Z,
  "id": 19,
  "name": "LeBron James",
  "surname": "paqueton",
}
`;
